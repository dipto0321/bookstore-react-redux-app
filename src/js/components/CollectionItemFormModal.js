import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import CollectionItemForm from './CollectionItemForm';

const CollectionItemFormModal = ({ addItemMode, switchAddItemMode }) => (
  <Modal
    isOpen={addItemMode}
    ariaHideApp={false}
    onRequestClose={switchAddItemMode}
    closeTimeoutMS={200}
  >
    <div className="modal-controls">
      <button
        type="button"
        className="cancel-btn"
        onClick={switchAddItemMode}
      >
        X
      </button>
    </div>
    <div className="modal-body">
      <CollectionItemForm />
    </div>
  </Modal>
);

CollectionItemFormModal.propTypes = {
  addItemMode: PropTypes.bool.isRequired,
  switchAddItemMode: PropTypes.func.isRequired,
};

export default CollectionItemFormModal;
