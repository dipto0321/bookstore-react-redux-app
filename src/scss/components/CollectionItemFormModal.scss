.ReactModalPortal>div {
  opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
  transition: 0.3s opacity ease-in-out;
  display: flex;
  justify-content: center;
  align-items: center;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
  opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
  opacity: 0;
}

.modal-controls {
  text-align: right;
}

.cancel-btn {
  width: 2rem;
  height: 2rem;
  text-align: center;
  background: $deep-red;
  color: rgb(223, 223, 192);
  font-weight: 700;
}