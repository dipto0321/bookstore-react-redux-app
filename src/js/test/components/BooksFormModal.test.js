import React from 'react';
import { shallow } from 'enzyme';
import CollectionItemFormModal from '../../components/CollectionItemFormModal';

test('CollectionItemFormModal should render correctly', () => {
  const wrapper = shallow(
    <CollectionItemFormModal
      addItemMode={false}
      switchAddItemMode={() => {}}
    />,
  );
  expect(wrapper).toMatchSnapshot();
});
