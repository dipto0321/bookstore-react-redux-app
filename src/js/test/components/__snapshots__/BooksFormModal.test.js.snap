// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItemFormModal should render correctly 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={200}
  isOpen={false}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal-controls"
  >
    <button
      className="cancel-btn"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div
    className="modal-body"
  >
    <ConnectFunction />
  </div>
</Modal>
`;
